database/rrdcalc.h

```diff
@@ -25,29 +35,42 @@ typedef enum {
     RRDCALC_FLAG_DISABLED                   = (1 << 7),
     RRDCALC_FLAG_SILENCED                   = (1 << 8),
     RRDCALC_FLAG_RUN_ONCE                   = (1 << 9),
-    RRDCALC_FLAG_FROM_TEMPLATE              = (1 << 10), // the rrdcalc has been created from a template
 } RRDCALC_FLAGS;
-
 void rrdcalc_flags_to_json_array(BUFFER *wb, const char *key, RRDCALC_FLAGS flags);
 
-typedef enum {
-    // This list uses several other options from RRDR_OPTIONS for db lookups.
-    // To add an item here, you need to reserve a bit in RRDR_OPTIONS.
-    RRDCALC_OPTION_NO_CLEAR_NOTIFICATION    = RRDR_OPTION_HEALTH_RSRVD1,
-} RRDCALC_OPTIONS;
-
 #define RRDCALC_ALL_OPTIONS_EXCLUDING_THE_RRDR_ONES (RRDCALC_OPTION_NO_CLEAR_NOTIFICATION)
 
-struct rrdcalc {
-    STRING *key;                    // the unique key in the host's rrdcalc_root_index
+struct rrd_alert_match {
+    bool enabled;
 
-    uint32_t id;                    // the unique id of this alarm
-    uint32_t next_event_id;         // the next event id that will be used for this alarm
+    bool is_template;
+    union {
+        STRING *chart;
+        STRING *context;
+    } on;
+
+    STRING *os;
+    STRING *host;
+    STRING *charts;                         // the charts that should be linked to (for templates)
+    STRING *plugin;                         // the plugin name that should be linked to
+    STRING *module;                         // the module name that should be linked to
+    STRING *host_labels;                    // the label read from an alarm file
+    STRING *chart_labels;                   // the chart label read from an alarm file
+
+    SIMPLE_PATTERN *os_pattern;
+    SIMPLE_PATTERN *host_pattern;
+    SIMPLE_PATTERN *charts_pattern;         // the simple pattern of charts
+    SIMPLE_PATTERN *plugin_pattern;         // the simple pattern of plugin
+    SIMPLE_PATTERN *module_pattern;         // the simple pattern of module
+    SIMPLE_PATTERN *host_labels_pattern;    // the simple pattern of labels
+    SIMPLE_PATTERN *chart_labels_pattern;   // the simple pattern of chart labels
+};
+void rrd_alert_match_cleanup(struct rrd_alert_match *am);
 
-    uuid_t config_hash_id;          // a predictable hash_id based on specific alert configuration
+struct rrd_alert_config {
```

Comment:
I think this should be moved under `health` as well.

Reply:
same as above